### PR TITLE
CSS Generator Cleanup

### PR DESCRIPTION
**Pull request by ludwiczakpawel**
## What type of PR is this? (check all applicable)

- [x] Refactor

## Description

This PR cleans up our CSS utility-first classes generator. I've removed unused (and unlikely to be used) classes. That should save us around ~4000k lines of pre-optimized CSS code. If any of these will ever be needed we can simply add them manually.

This could be considered either as preparation to fully adopt official version of TailwindCSS OR preparation to fully drop TailwindCSS **idea** 🤷 . Nevertheless it's optimization.

## QA Instructions, Screenshots, Recordings

This should **not** have any affect on UI - all the classes were unused.

## Added/updated tests?

- [x] No

## [Forem core team only] How will this change be communicated?

- [x] No

### CODE REVIEW

File: app/assets/stylesheets/config/_generator.scss [hunk 1]
```diff
@@ -60,79 +60,18 @@
     'border-radius',
     ('default': var(--radius), '0': 0, 'full': 9999px),
     (),
-    true
```
Comment (nickytonline) [inline app/assets/stylesheets/config/_generator.scss:63]: Are all these key/value pairs being removed because we literally never use these classes at the moment in the codebase, i.e. manual CSS purge?

File: app/assets/stylesheets/config/_generator.scss [hunk 2]
```diff
@@ -184,23 +123,19 @@
     (),
     true
   ),
-  ('grow', 'flex-grow', ('0': 0, '1': 1), (), true),
-  ('shrink', 'flex-shrink', ('0': 0, '1': 1), (), true),
+  ('grow', 'flex-grow', ('0': 0, '1': 1), (), false),
+  ('shrink', 'flex-shrink', ('0': 0, '1': 1), (), false),
   ('order', 'order', ('first': -9999, 'last': 9999, '0': 0), (), true),
-  ('gap', 'grid-gap', ('0': 0), $spacing-units, true),
+  ('gap', 'gap', ('0': 0), $spacing-units, true),
```
Comment (nickytonline) [inline app/assets/stylesheets/config/_generator.scss:129]: Just putting a note here that `grid-gap` is deprecated in favour of `gap`.

File: app/javascript/article-form/components/Help/EditorFormattingHelp.jsx [hunk 3]
```diff
@@ -78,7 +78,7 @@ export const EditorFormattingHelp = ({ openModal }) => (
               <tr>
                 <td className="ff-monospace">&gt; quoted text</td>
                 <td>
-                  <span className="pl-2 border-0 border-solid border-l-4 border-base-50">
+                  <span className="pl-2 border-0 border-solid border-l-2 border-base-50">
```
Comment (nickytonline) [inline app/javascript/article-form/components/Help/EditorFormattingHelp.jsx:81]: nitpick (non-blocking): Move this small change to a separate PR as it's not related to the CSS generator cleanup.

**Review by nickytonline — COMMENTED** [review]
This looks good, but we definitely need a few folks to navigate the site to ensure nothing major is broken UI wise.

File: app/assets/stylesheets/config/_generator.scss (hunk 1 above)
Comment (ludwiczakpawel) [reply app/assets/stylesheets/config/_generator.scss:63]: Correct. And I removed them because:
- in case we end up using Tailwind library we gonna use PurgeCSS anyway so those would be gone 🤷 
- in case we don't end up using Tailwind library, these classes weren't used anyway. If we ever need one, we can always manually add.

File: app/javascript/article-form/components/Help/EditorFormattingHelp.jsx (hunk 3 above)
Comment (ludwiczakpawel) [reply app/javascript/article-form/components/Help/EditorFormattingHelp.jsx:81]: it was the only element using that `border-l-4` class and since I was reducing other borders (`border-`, `border-r-`, `border-t-`, `border-b-` from unnecessary [sizes/widths](https://github.com/forem/forem/pull/15284/files#diff-73bcc9a8dbf65b2d301e5d0db91bb0e471cb1372f8a23dcd9c02a86418309b24L256-L295) (`...3`, `...4`, `...8`) I decided to sneak that one in as well. Otherwise I wouldn't do it. FWIW, this almost hasn't affected overall aesthetics - it's just thinner border but not significantly.

**Review by nickytonline — APPROVED** [review]
I did a smoke test of the site, and nothing stands out in the UI as being broken from a stylistic point of view. I'll let @aitchiss take it for a spin as well in case I missed something.

**Review by aitchiss — APPROVED** [review]
Took it for a spin on Safari/Firefox/Chrome at various viewport sizes and didn't notice anything rogue - LGTM 👍
